src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlConnectionPoolGroupProviderInfo.cs

```diff
@@ -55,14 +59,47 @@ internal void AliasCheck(string server)
                     }
                     else if (_alias != server)
                     {
-                        SqlClientEventSource.Log.TryTraceEvent("<sc.SqlConnectionPoolGroupProviderInfo|INFO> alias change detected. Clearing PoolGroup");
+                        SqlClientEventSource.Log.TryTraceEvent("SqlConnectionPoolGroupProviderInfo.AliasCheck | Info | Alias change detected. Clearing PoolGroup.");
                         base.PoolGroup.Clear();
                         _alias = server;
                     }
                 }
             }
         }
 
+        internal void FailoverCheck(SqlInternalConnection connection, bool actualUseFailoverPartner, SqlConnectionString userConnectionOptions, string actualFailoverPartner)
+        {
+            if (UseFailoverPartner != actualUseFailoverPartner)
+            {
+                SqlClientEventSource.Log.TryTraceEvent("SqlConnectionPoolGroupProviderInfo.FailoverCheck | Info | Failover detected. Failover partner '{0}'. Clearing PoolGroup", actualFailoverPartner);
+                base.PoolGroup.Clear();
+                _useFailoverPartner = actualUseFailoverPartner;
+            }
+            // Only construct a new permission set when we're connecting to the
+            // primary data source, not the failover partner.
+            if (!_useFailoverPartner && _failoverPartner != actualFailoverPartner)
+            {
+                // NOTE: we optimistically generate the permission set to keep
+                //       lock short, but we only do this when we get a new
+                //       failover partner.
+
+#if NETFRAMEWORK
+                System.Security.PermissionSet failoverPermissionSet = CreateFailoverPermission(userConnectionOptions, actualFailoverPartner);
+#endif
+                lock (this)
+                {
+                    if (_failoverPartner != actualFailoverPartner)
+                    {
+                        _failoverPartner = actualFailoverPartner;
+#if NETFRAMEWORK
+                        _failoverPermissionSet = failoverPermissionSet;
+#endif
+                    }
+                }
+            }
+        }
+
+#if NETFRAMEWORK
         private System.Security.PermissionSet CreateFailoverPermission(SqlConnectionString userConnectionOptions, string actualFailoverPartner)
```

Comment:
Plus at line 140.
```suggestion
        private ermissionSet CreateFailoverPermission(SqlConnectionString userConnectionOptions, string actualFailoverPartner)
```

Reply:
just realize I already applied this change already in my commit, but because I was switching between 2 PR branches, I accidentally pushed it to the wrong one.